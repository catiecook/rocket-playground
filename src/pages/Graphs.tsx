import React from 'react';
import Rockets from '../components/Rockets';
import Year from '../components/Year';
import Cost from '../components/Cost';
import '../styles/Graph.css';

// REFACTOR NOTE:
// This is where the apollo / gql call should be executed
// that way the call is made ont time, and then data is parced and sent
// into components via props.
class Graphs extends React.Component {

  render() {
    return (
      <div className="container">
        <Rockets />
        <Year />
        <Cost />
      </div>
    )
  }
}

export default Graphs;
