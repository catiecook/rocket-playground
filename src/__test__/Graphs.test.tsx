import React from "react";
import ReactDOM from 'react-dom';
import { fireEvent, waitForElement } from "@testing-library/react";
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Graphs from "../pages/Graphs";

Enzyme.configure({ adapter: new Adapter() })


// snapshot tests
it('Graphs view renders as expected', () => {
  const wrapper = shallow(<Graphs />)
  expect(toJson(wrapper)).toMatchSnapshot();
});

describe("<Graphs />", () => {
  test("renders Graphs  view", async () => {
    const wrapper = shallow(<Graphs />);
    expect(wrapper.find("container"));
  });
});
