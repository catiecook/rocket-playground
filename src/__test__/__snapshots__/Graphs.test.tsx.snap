// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphs view renders as expected 1`] = `
<div
  className="container"
>
  <Rockets />
  <Year />
</div>
`;
